server-base/src/main/java/org/apache/kylin/rest/service/CubeService.java

```diff
@@ -610,6 +610,32 @@ public CubeInstance rebuildLookupSnapshot(CubeInstance cube, String segmentName,
         return cube;
     }
 
+    public CubeInstance deleteSegmentById(CubeInstance cube, String uuid) throws IOException {
+        aclEvaluate.checkProjectWritePermission(cube);
+        Message msg = MsgPicker.getMsg();
+
+        CubeSegment toDelete = null;
+
+        toDelete = cube.getSegmentById(uuid);
+
+        if (toDelete == null) {
+            throw new BadRequestException(String.format(Locale.ROOT, msg.getSEG_NOT_FOUND(), uuid));
+        }
+
+        if (cube.getStatus() == RealizationStatusEnum.DISABLED
+                || (cube.getStatus() == RealizationStatusEnum.READY && toDelete.getStatus() != SegmentStatusEnum.READY)) {
```

Comment:
When a segment with `NEW` status has related running jobs, I think it shouldn't be able to be deleted. Method `CubeService.isOrphonSegment(CubeInstance cube, String segId)` can help to judge whether a segment has related jobs.

Reply:
good catch!  fix.